test/lit/wat-kitchen-sink.wast

```diff
@@ -701,6 +701,112 @@
   drop
  )
 
+ ;; CHECK:      (func $block (type $void)
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (block $l
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:   (nop)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $block
+  nop
+  block $l (type $void)
+   nop
+   nop
+   nop
+  end $l
+ )
+
+ ;; CHECK:      (func $block-folded (type $void)
+ ;; CHECK-NEXT:  (local $scratch (i32 i32))
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT:  (local.set $scratch
+ ;; CHECK-NEXT:   (block $l (result i32 i32)
+ ;; CHECK-NEXT:    (nop)
+ ;; CHECK-NEXT:    (nop)
+ ;; CHECK-NEXT:    (unreachable)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (tuple.extract 0
+ ;; CHECK-NEXT:    (local.get $scratch)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
```

Comment:
We don't support using the results yet, and so that part is not tested and we can only show drops here, is that right? (If it isn't, I missed it in the code...)

Reply:
No, we support using the results of blocks just fine. The drops here could be any instruction. Once the inner block is finished and its scope is popped, it is pushed into the outer scope just like any other instruction. See the calls to `ctx.appendInstr()` inside `instrs()`.